frame/dvm/ethereum/src/lib.rs

```diff
@@ -302,13 +302,13 @@ pub mod pallet {
 	}
 
 	#[pallet::event]
-	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	#[pallet::generate_deposit(pub fn deposit_event)]
 	/// Ethereum pallet events.
-	pub enum Event {
+	pub enum Event<T: Config> {
 		/// An ethereum transaction was successfully executed. \[from, to/contract_address, transaction_hash, exit_reason\]
 		Executed(H160, H160, H256, ExitReason),
-		/// DVM Transfer succeeded. \[from, to, value\]
-		DVMTransfer(H160, H160, U256),
+		/// DVM Transfer succeeded. \[AddrWrapper(from), AddrWrapper(to), value\]
+		DVMTransfer(AddrWrapper<T::AccountId>, AddrWrapper<T::AccountId>, U256),
```

Comment:
Why not directly use `T::AccountId` ? `AddrWrapper` introduce extra enum type here and unnecessary match in implementations.

Reply:
Actually `AddrWrapper` make `account_basic` implementation much easier. We do not need to split some functions into two versions, `evm_transfer` and `substrate_transfer` anymore. And for the application developer, this event is much more readable in my view.

Reply:
First, `AddrWrapper ` will make the user of this event pay extra attention to decode it, it break consistency of subscan current account interpreter, subscan will need to pay more attention for the first/byte encoded by the enum.

Second, to be frankly, I think using two method here is much more simpler and easy to read here, I don't need to knwo what's the concept of AddrWrapper and the `match switch` bad pattern.

Reply:
> First, AddrWrapper will make the user of this event pay extra attention to decode it, it break consistency of subscan current account interpreter/render, subscan will need to pay more attention for the first byte encoded by the enum.

After V14, no JSON type is needed, the subscan side can decode this struct directly without extra effort. If using `AddrWrapper`, they don't need to do conversation from AccountId to H160. Otherwise, they need this convert.